core/src/SYCL/Kokkos_SYCL_Parallel_Range.hpp

```diff
@@ -256,7 +265,9 @@ class Kokkos::Impl::ParallelFor<FunctorType, Kokkos::MDRangePolicy<Traits...>,
             .exec_range();
       });
     });
-    q.ext_oneapi_submit_barrier(std::vector<sycl::event>{parallel_for_event});
+    // This barrier would prevent q.memcpy for subsequent kernels from being
+    // brought forward in time.
+    // q.ext_oneapi_submit_barrier(std::vector<sycl::event>{parallel_for_event});
```

Comment:
I don't see how this could be safe to do in general. Due to an assumed compiler regression, we are currently creating all `sycl::queues` as in-order, and thus adding the synchronization event here shouldn't really have any effect but I want to get rid of this restriction as quickly as possible.
```suggestion
    q.ext_oneapi_submit_barrier(std::vector<sycl::event>{parallel_for_event});
```

Reply:
OK, done.